crates/relayer/src/chain/client.rs

```diff
@@ -24,10 +24,18 @@ impl ClientSettings {
         // Currently, only Tendermint chain pairs are supported by
         // ForeignClient::build_create_client_and_send. Support for
         // heterogeneous chains is left for future revisions.
-        ClientSettings::Tendermint(cosmos::client::Settings::for_create_command(
-            options,
-            src_chain_config,
-            dst_chain_config,
-        ))
+        //
+        // TODO: extract Tendermint-related configs into a separate substructure
+        // that can be used both by CosmosSdkConfig and configs for nonSDK chains.
```

Comment:
It seems like the right move here is to split out the Tendermint-related fields on the config into a Tendermint-specific config structure. Then that structure can be a field on both `CosmosSdkConfig` and other Tendermint-using chains (like Penumbra or Namada).